appcelerator/titanium_mobile
# feat(android): add Ti.UI.ButtonBar

**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-25954

**Summary:**
- This used to be an iOS only feature implemented as a segmented control. (See: [link](https://developer.apple.com/design/human-interface-guidelines/ios/controls/segmented-controls))
- On Android, it's implemented via material design outline buttons connected together.
- Note that Android supports a button with an image and an icon. iOS does not support both.

**Screenshot:**
![ButtonBar-Android](https://user-images.githubusercontent.com/26234558/108463238-e0c6fa80-7232-11eb-9d1a-4c8d6654eb0c.png)

---
**Test:**
1. Create a Classic Titanium project. (Use the template that creates a TabGroup.)
2. Use the `ButtonBarTest.js` as your `app.js` from [TIMOB-25954](https://jira.appcelerator.org/browse/TIMOB-25954).
3. Add the below <application/> element to the "tiapp.xml" file.
4. Build and run on Android.
5. Verify the app looks like the above screenshot.
6. Tap on the 2nd button of each bar and verify the following gets logged.
`@@@ button click index: 1`

```xml
<ti:app>
	<android xmlns:android="http://schemas.android.com/apk/res/android">
		<manifest>
			<application android:theme="@style/Theme.MaterialComponents.Light.DarkActionBar"/>
		</manifest>
	</android>
</ti:app>
```
